### ISSUE TITLE
ldc-1.18.0 source code build fails in Homebrew

### ISSUE THREAD

**Issue by denniscote** [ldc-developers/ldc#3196]
Hi,

I was trying to update the Homebrew formula for ldc to 1.18.0 but the source code build is failing for this version. I have successfully updated the formula in the past. 

I have copied the output of the test build, but I am not familiar with cmake, so I don't know what is going wrong here. The last two lines of output suggest to view some log files, but there is no ldc-20191020... directory in my /tmp/ directory, so I have no other information to go on.

I have also upgraded macOS from Mojave to Catalina since I last updated the homebrew formula for ldc, so that may be part of the issue.

Do you have any idea what might be going on? Any recommendadtions for things to check or ideas to try to correct the problem?

TIA 
Dennis Cote

````
bash-3.2$ brew install --build-from-source ldc
==> Downloading https://github.com/ldc-developers/ldc/releases/download/v1.18.0/
Already downloaded: /Users/dennis/Library/Caches/Homebrew/downloads/f792bec328d149c0b7541ff1c534d65c8d8b5e5c5a089359f6407a4398c44078--ldc-1.18.0-src.tar.gz
==> Downloading https://github.com/ldc-developers/ldc/releases/download/v1.18.0/
Already downloaded: /Users/dennis/Library/Caches/Homebrew/downloads/6753c2dee7a3cf2b7659da9f58a6753bcfdeb50b69f6074cf57d7790f76b9e57--ldc2-1.18.0-osx-x86_64.tar.xz
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG
Last 15 lines from /Users/dennis/Library/Logs/Homebrew/ldc/01.cmake:
  Targets Defined: FileCheck;not

  Targets not yet defined:
  LLVMDemangle;LLVMSupport;LLVMTableGen;llvm-tblgen;LLVMCore;LLVMFuzzMutate;LLVMIRReader;LLVMCodeGen;LLVMSelectionDAG;LLVMAsmPrinter;LLVMMIRParser;LLVMGlobalISel;LLVMBinaryFormat;LLVMBitReader;LLVMBitWriter;LLVMBitstreamReader;LLVMTransformUtils;LLVMInstrumentation;LLVMAggressiveInstCombine;LLVMInstCombine;LLVMScalarOpts;LLVMipo;LLVMVectorize;LLVMObjCARCOpts;LLVMCoroutines;LLVMLinker;LLVMAnalysis;LLVMLTO;LLVMMC;LLVMMCParser;LLVMMCDisassembler;LLVMMCA;LLVMObject;LLVMObjectYAML;LLVMOption;LLVMRemarks;LLVMDebugInfoDWARF;LLVMDebugInfoGSYM;LLVMDebugInfoMSF;LLVMDebugInfoCodeView;LLVMDebugInfoPDB;LLVMSymbolize;LLVMExecutionEngine;LLVMInterpreter;LLVMJITLink;LLVMMCJIT;LLVMOrcJIT;LLVMRuntimeDyld;LLVMTarget;LLVMAArch64CodeGen;LLVMAArch64AsmParser;LLVMAArch64Disassembler;LLVMAArch64Desc;LLVMAArch64Info;LLVMAArch64Utils;LLVMAMDGPUCodeGen;LLVMAMDGPUAsmParser;LLVMAMDGPUDisassembler;LLVMAMDGPUDesc;LLVMAMDGPUInfo;LLVMAMDGPUUtils;LLVMARMCodeGen;LLVMARMAsmParser;LLVMARMDisassembler;LLVMARMDesc;LLVMARMInfo;LLVMARMUtils;LLVMBPFCodeGen;LLVMBPFAsmParser;LLVMBPFDisassembler;LLVMBPFDesc;LLVMBPFInfo;LLVMHexagonCodeGen;LLVMHexagonAsmParser;LLVMHexagonDisassembler;LLVMHexagonDesc;LLVMHexagonInfo;LLVMLanaiCodeGen;LLVMLanaiAsmParser;LLVMLanaiDisassembler;LLVMLanaiDesc;LLVMLanaiInfo;LLVMMipsCodeGen;LLVMMipsAsmParser;LLVMMipsDisassembler;LLVMMipsDesc;LLVMMipsInfo;LLVMMSP430CodeGen;LLVMMSP430Desc;LLVMMSP430Info;LLVMMSP430AsmParser;LLVMMSP430Disassembler;LLVMNVPTXCodeGen;LLVMNVPTXDesc;LLVMNVPTXInfo;LLVMPowerPCCodeGen;LLVMPowerPCAsmParser;LLVMPowerPCDisassembler;LLVMPowerPCDesc;LLVMPowerPCInfo;LLVMRISCVCodeGen;LLVMRISCVAsmParser;LLVMRISCVDisassembler;LLVMRISCVDesc;LLVMRISCVInfo;LLVMRISCVUtils;LLVMSparcCodeGen;LLVMSparcAsmParser;LLVMSparcDisassembler;LLVMSparcDesc;LLVMSparcInfo;LLVMSystemZCodeGen;LLVMSystemZAsmParser;LLVMSystemZDisassembler;LLVMSystemZDesc;LLVMSystemZInfo;LLVMWebAssemblyCodeGen;LLVMWebAssemblyAsmParser;LLVMWebAssemblyDisassembler;LLVMWebAssemblyDesc;LLVMWebAssemblyInfo;LLVMX86CodeGen;LLVMX86AsmParser;LLVMX86Disassembler;LLVMX86Desc;LLVMX86Info;LLVMX86Utils;LLVMXCoreCodeGen;LLVMXCoreDisassembler;LLVMXCoreDesc;LLVMXCoreInfo;LLVMAsmParser;LLVMLineEditor;LLVMProfileData;LLVMCoverage;LLVMPasses;LLVMTextAPI;LLVMDlltoolDriver;LLVMLibDriver;LLVMXRay;LLVMWindowsManifest;llvm-PerfectShuffle;count;yaml-bench;PollyISL;PollyPPCG;Polly;LLVMPolly;LTO;llvm-ar;llvm-config;llvm-lto;llvm-profdata;SampleAnalyzerPlugin;CheckerDependencyHandlingAnalyzerPlugin;CheckerOptionHandlingAnalyzerPlugin;lit-cpuid;bugpoint;dsymutil;llc;lli-child-target;lli;llvm-as;llvm-bcanalyzer;llvm-c-test;llvm-cat;llvm-cfi-verify;llvm-cov;llvm-cvtres;llvm-cxxdump;llvm-cxxfilt;llvm-cxxmap;llvm-diff;llvm-dis;llvm-dwarfdump;llvm-dwp;llvm-elfabi;llvm-exegesis;llvm-extract;llvm-jitlink;llvm-link;llvm-lipo;llvm-lto2;llvm-mc;llvm-mca;llvm-modextract;llvm-mt;llvm-nm;llvm-objcopy;llvm-objdump;llvm-opt-report;llvm-pdbutil;llvm-rc;llvm-readobj;llvm-rtdyld;LLVM;llvm-size;llvm-split;llvm-stress;llvm-strings;llvm-symbolizer;llvm-undname;llvm-xray;obj2yaml;opt;Remarks;sancov;sanstats;verify-uselistorder;yaml2obj


Call Stack (most recent call first):
  /usr/local/Cellar/llvm/9.0.0/lib/cmake/llvm/LLVMConfig.cmake:238 (include)
  runtime/jit-rt/DefineBuildJitRT.cmake:54 (include)
  runtime/CMakeLists.txt:776 (build_jit_runtime)


-- Configuring incomplete, errors occurred!
See also "/tmp/ldc-20191020-87282-1x4w634/ldc-1.18.0-src/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/ldc-20191020-87282-1x4w634/ldc-1.18.0-src/build/CMakeFiles/CMakeError.log".

READ THIS: https://docs.brew.sh/Troubleshooting

**Comment by kinke** [comment]
This is a duplicate of #3079 - use `-DLDC_DYNAMIC_COMPILE=False` in the CMake cmdline.

@Hardcode84: This CMake hack is also an issue with new Ubuntu 19.10 and its LLVM 9. #3187 doesn't suffice, as can be seen by recent CircleCI runs:
```
CMake Error at /usr/lib/llvm-9/lib/cmake/llvm/LLVMExports.cmake:1323 (message):
  The imported target "yaml-bench" references the file

     "/usr/lib/llvm-9/bin/yaml-bench"

  but this file does not exist.
```

**Comment by denniscote** [comment]
Hi

> On Oct 20, 2019, at 1:35 PM, Martin Kinkelin <notifications@github.com> wrote:
> 
> This is a duplicate of #3079 <https://github.com/ldc-developers/ldc/issues/3079> - use -DLDC_DYNAMIC_COMPILE=False in the CMake cmdline.

I tried adding this define to the cmake command but I get the same result. The cmake log file of the run is attached.

**Comment by kinke** [comment]
Unfortunately, there's a 2nd location including the LLVM cmake files, so you'll need https://github.com/ldc-developers/ldc/pull/3198.

[Mail attachments don't show up in the web interface.]

**Comment by denniscote** [comment]
OK, then I’ll have to wait until you release an updated version 1.18.1 of ldc source files since Homebrew builds from the released archive files on Github. It’s better to leave Homebrew at 1.17.0 than move to 1.18.0 that won’t build.

Thanks for all your efforts on ldc!

Have a good day.

**Comment by kinke** [comment]
> until you release an updated version 1.18.1

Not sure whether that's worth it. These CMake issues have surfaced with custom builds of LLVM 9 (and of course no issues with LLVM built by ourselves, on a multitude of platforms, and no problem with the official prebuilt LLVM 9 on Ubuntu either), so I'd guess you could still release a new Homebrew version with previous LLVM 8.0.1. The offending lines were already there in LLVM v1.17 and earlier.

**Comment by denniscote** [comment]
> On Oct 22, 2019, at 1:07 PM, Martin Kinkelin <notifications@github.com> wrote:
> 
> you could still release a new Homebrew version with previous LLVM 8.0.1

That won’t work very well since most Homebrew users automatically update their installed software. Llvm 9.0.0 was installed on my machine 09-24, probably a day or two after it was released. Requiring a temporary installation of llvm 8.0.1 just to build ldc from source (which is not normally done, but must be supported for a new formula in Homebrew) seems excessive. Homebrew users will typically install a precompiled binary version of ldc, but Homebrew requires that those binaries are built from source on their servers using the current “formula", a script written in ruby. I was testing the formula, modified for ldc 1.18.0, when I ran into the problem.
